ksqldb-functional-tests/src/test/resources/query-validation-tests/row-id.json

```diff
@@ -0,0 +1,227 @@
+        "INSERT INTO a SELECT id, ROWID from b EMIT CHANGES;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Reserved column name in select: `ROWID`. Please remove or alias the column."
+      }
+    },
+    {
+      "name": "Select ROWID columns with aliases - stream",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT id, ROWID AS ri, ROWTIME AS rt FROM input;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp":  7}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"RI": "AAAAAQ==", "RT": 5}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 2, "value": {"RI": "AAAAAg==", "RT": 6}, "timestamp": 6},
+        {"topic":  "OUTPUT", "key": 3, "value": {"RI": "AAAAAw==", "RT": 7}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, RI BYTES, RT BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "Select ROWID with aliases - table",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE TABLE input (id INT PRIMARY KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE output AS SELECT id, ROWID AS ri FROM input;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"foo": "a"}, "timestamp": 5},
+        {"topic":  "test_topic", "key": 2, "value": {"foo": "b"}, "timestamp": 6},
+        {"topic":  "test_topic", "key": 3, "value": {"foo": "c"}, "timestamp":  7}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"RI": "AAAAAQ=="}, "timestamp": 5},
+        {"topic":  "OUTPUT", "key": 2, "value": {"RI": "AAAAAg=="}, "timestamp": 6},
+        {"topic":  "OUTPUT", "key": 3, "value": {"RI": "AAAAAw=="}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "table", "schema": "ID INT KEY, RI BYTES"}
+        ]
+      }
+    },
+    {
+      "name": "join on ROWID",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM a (id STRING KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM b (id STRING KEY, bar STRING) WITH (kafka_topic='test_topic_2', value_format='JSON');",
+        "CREATE STREAM output AS SELECT foo, bar, a.ROWID AS ria, b.ROWID as rib FROM a JOIN b WITHIN 5 SECONDS ON a.ROWID = b.ROWID;"
```

Comment:
Why would a user ever want to do this, given that the streams already have key columns defined? Is this join statement expected to work even if the streams don't have key columns defined? If so, I think we'll want work in the planner to not repartition on joins on ROWID. Currently it's being detected as a non-key column so both these streams are getting repartitioned, which is very wasteful.